src/OpenLoco/src/Scenario.cpp

```diff
@@ -531,47 +525,40 @@ namespace OpenLoco::Scenario
             }
             ObjectManager::selectObjectFromIndex(ObjectManager::SelectObjectModes::defaultDeselect, *oldCurrency.object._header, getInUseSelectedObjectFlags(), _objectSelectionMeta);
         }
-        ObjectManager::selectObjectFromIndex(ObjectManager::SelectObjectModes::defaultSelect, preferredCurreny, getInUseSelectedObjectFlags(), _objectSelectionMeta);
+        if (!ObjectManager::selectObjectFromIndex(ObjectManager::SelectObjectModes::defaultSelect, preferredCurreny, getInUseSelectedObjectFlags(), _objectSelectionMeta))
+        {
+            // Failed so reselect the old currency and give up
+            ObjectManager::selectObjectFromIndex(ObjectManager::SelectObjectModes::defaultSelect, *oldCurrency.object._header, getInUseSelectedObjectFlags(), _objectSelectionMeta);
```

Comment:
Should we unset the currency preferences at this point too?

Reply:
I was thinking perhaps not. If you are just switching out objects for one time i think you would want it to return back when you have your objects

Reply:
But if you think that doesn't make sense then can change it.

Reply:
Sounds sensible for that use case. I'm just worried it'll have unintended side effects for the company face selection window, for example. Is that working well like this?

Reply:
It will get unset if you open the options window whilst in this state. Same goes for currency.

Reply:
Alright, let's leave it as-is then.